ui/pages/Login/components/Login.jsx

```diff
@@ -12,6 +16,48 @@ const FIELDS = [
   { name: 'password', label: 'Password', type: 'password', validate: validators.required },
 ]
 
+const googleLogin = () => {
+  return new HttpRequestHelper('/api/login_google',
+    (responseJson) => {
+      // Redirect to google auth website
+      const width = 600
+      const height = 800
+      const left = (window.screen.width - width) / 2
+      const top = (window.screen.height - height) / 2
+      const params = `scrollbars=no,status=no,location=no,toolbar=no,menubar=no,
+        width=${width},height=${height},left=${left},top=${top}`
+      const win = window.open(responseJson.data, 'Google Sign In', params)
+      win.focus()
+    },
+    (e) => {
+      throw new SubmissionError({ _error: [e.message] })
+    },
+  ).get()
+}
+
+export const OAuth2Callback = ({ location }) => {
+  // Send the authentication results to the backend to finish the logging-in procedure
+  new HttpRequestHelper('/api/login_oauth2callback',
```

Comment:
this function should be in the reducers file

Reply:
The `OAuth2Callback` is a react component, not a real callback function. It is the web page that the Google auth server will redirect to after the user is authenticated. The Google sample code uses the name `OAuth2Callback` and I left it as is.